clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java

```diff
@@ -60,13 +60,16 @@
     private TachyonFileSystemFactory() {} // to prevent initialization
 
     public static synchronized TachyonFileSystem get() {
-      if (sTachyonFileSystem == null) {
-        boolean enableLineage = ClientContext.getConf().getBoolean(Constants.USER_LINEAGE_ENABLED);
-        sTachyonFileSystem =
-            enableLineage ? TachyonLineageFileSystem.get() : new TachyonFileSystem();
-      }
-      return sTachyonFileSystem;
+      boolean enableLineage = ClientContext.getConf().getBoolean(Constants.USER_LINEAGE_ENABLED);
+      return enableLineage ? TachyonLineageFileSystem.get() : TachyonFileSystem.get();
+    }
+  }
+
+  static synchronized TachyonFileSystem get() {
+    if (sTachyonFileSystem == null) {
+      sTachyonFileSystem = new TachyonFileSystem();
```

Comment:
Ahhh, I get it now. sTachyonFileSystem will always be a TachyonFileSystem(), where as before it could also be a TachyonLineageFileSystem(). It makes more sense now.